src/main/kotlin/uk/gov/dluhc/printapi/rds/entity/Certificate.kt

```diff
@@ -0,0 +1,118 @@
+package uk.gov.dluhc.printapi.rds.entity
+
+import org.hibernate.Hibernate
+import org.hibernate.annotations.GenericGenerator
+import org.hibernate.annotations.Type
+import org.hibernate.annotations.UpdateTimestamp
+import org.springframework.data.annotation.LastModifiedBy
+import org.springframework.data.jpa.domain.support.AuditingEntityListener
+import uk.gov.dluhc.printapi.database.entity.SourceType
+import uk.gov.dluhc.printapi.database.entity.Status
+import uk.gov.dluhc.printapi.rds.repository.UUIDCharType
+import uk.gov.dluhc.printapi.rds.repository.UseExistingOrGenerateUUID
+import java.time.Instant
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+import javax.persistence.CascadeType
+import javax.persistence.Entity
+import javax.persistence.EntityListeners
+import javax.persistence.EnumType
+import javax.persistence.Enumerated
+import javax.persistence.FetchType
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+import javax.persistence.JoinColumn
+import javax.persistence.OneToMany
+import javax.persistence.Table
+import javax.persistence.Version
+import javax.validation.constraints.NotNull
+import javax.validation.constraints.Size
+
+@Table
+@Entity
+@EntityListeners(AuditingEntityListener::class)
+class Certificate(
+
+    @Id
+    @Type(type = UUIDCharType)
+    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = UseExistingOrGenerateUUID.NAME)
+    var id: UUID? = null,
+
+    @NotNull
+    @Size(max = 20)
+    var vacNumber: String? = null,
+
+    @NotNull
+    @Size(max = 20)
+    @Enumerated(EnumType.STRING)
+    var sourceType: SourceType? = null,
+
+    @NotNull
+    @Size(max = 255)
+    var sourceReference: String? = null,
+
+    @Size(max = 255)
+    var applicationReference: String? = null,
+
+    @NotNull
+    var applicationReceivedDateTime: OffsetDateTime? = null,
+
+    @NotNull
+    @Size(max = 255)
+    var issuingAuthority: String? = null,
+
+    @NotNull
+    var issueDate: LocalDate = LocalDate.now(),
+
+    @NotNull
+    var suggestedExpiryDate: LocalDate = issueDate.plusYears(10),
+
+    @NotNull
+    @Size(max = 50)
+    @Enumerated(EnumType.STRING)
+    var status: Status = Status.PENDING_ASSIGNMENT_TO_BATCH,
```

Comment:
@Neil-Massey , looking into the current implementation, the status needs to be inferred from the print request statuses by taking the latest status.
 
Do you agree?

For the context, this is the current implementation with dynamodb:

<pre>
    var status: Status?
        @DynamoDbSecondaryPartitionKey(indexNames = [STATUS_BATCH_ID_INDEX_NAME])
        get() = printRequestStatuses?.sortedBy { it.eventDateTime }?.last()?.status
        @Deprecated(
            """
            Programmatically setting the status property is not supported and will have no effect.
            The status property and its setter are provided so that dynamodb persists a status attribute, where the value
            is the last element from the printRequestStatuses list (element with the most recent eventDateTime).
            To set the status add a new PrintRequestStatus to the printRequestStatuses list.
            """
        )
        set(_) { /* no-op setter */ }
</pre>

Reply:
Yes - It's actually a little bit more involved now as we have Certificate -> List<Requests> -> List<Statuses>
This PR was mainly around the Liquibase and Entities... but you're right how the status is to be managed needs updating...

Reply:
My understanding was when we completely migrate to `mysql` then we anyways need to revisit the bits done using `DynamoDB`. So in the scope of current PR this may not be needed.

Reply:
Updated to set the correct status when saving the certificate.  
Presumably most of the task processing will be based on the Certificate status (and not the PrintRequest) - although the events are against the individual request.